src/coreclr/jit/lowerxarch.cpp

```diff
@@ -3693,6 +3720,70 @@ void Lowering::LowerHWIntrinsicToScalar(GenTreeHWIntrinsic* node)
         LowerNode(cast);
     }
 }
+
+//----------------------------------------------------------------------------------------------
+// Lowering::LowerAndOpToResetLowestSetBit: Lowers a tree AND(X, ADD(X, -1) to HWIntrinsic::ResetLowestSetBit
+//
+// Arguments:
+//    andNode - GT_AND node of integral type
+//
+// Return Value:
+//    Returns the replacement node if one is created else nullptr indicating no replacement
+//
+GenTree* Lowering::LowerAndOpToResetLowestSetBit(GenTreeOp* andNode)
+{
+    assert(andNode->OperIs(GT_AND) && varTypeIsIntegral(andNode));
+
+    GenTree* op1 = andNode->gtGetOp1();
+    GenTree* op2 = andNode->gtGetOp2();
+    if (op1->OperIs(GT_LCL_VAR) && op2->OperIs(GT_ADD) && !comp->lvaGetDesc(op1->AsLclVar())->IsAddressExposed())
+    {
+        GenTree* addOp1 = op2->gtGetOp1();
+        GenTree* addOp2 = op2->gtGetOp2();
+        if (addOp2->IsIntegralConst(-1) && addOp1->OperIs(GT_LCL_VAR) &&
+            (addOp1->AsLclVar()->GetLclNum() == op1->AsLclVar()->GetLclNum()) &&
+            ((op1->TypeIs(TYP_LONG) && comp->compOpportunisticallyDependsOn(InstructionSet_BMI1_X64)) ||
+             comp->compOpportunisticallyDependsOn(InstructionSet_BMI1)))
+        {
+            LIR::Use use;
+            if (BlockRange().TryGetUse(andNode, &use))
+            {
+                // If we allow the use of BSLR where the parent is NE or EQ it will generate "blsr, test"
+                // if we prevent the use of BSLR here then "OptimizeConstCompare" may generate TEST_NE or
+                //   TEST_EQ leading to a cheaper "lea, test"
+                GenTree* user = use.User();
+                if (!(user->OperIs(GT_EQ, GT_NE) && user->gtGetOp2()->IsIntegralConst()))
+                {
+                    GenTreeHWIntrinsic* blsrNode =
+                        comp->gtNewScalarHWIntrinsicNode(andNode->TypeGet(), op1,
+                                                         andNode->TypeIs(TYP_INT)
+                                                             ? NamedIntrinsic::NI_BMI1_ResetLowestSetBit
+                                                             : NamedIntrinsic::NI_BMI1_X64_ResetLowestSetBit);
+
+                    JITDUMP("Lower: optimize AND(X, ADD(X, -1)): ");
```

Comment:
have you checked the DOTNET_JITDUMP with these logs, you might need to add \n inside JITDUMP and a `JITDUMP("\n") after `DISPNODE(blsrNode);`

Reply:
There are no uses of a newline only jitdump after a dispnode so i don't think that's needed. I will add a newline at the end of the jitdumps.